File: src/java/org/elastos/hive/connection/auth/AuthAPI.java
package org.elastos.hive.connection.auth;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;

public interface AuthAPI {
	@POST("/api/v2/did/signin")
	Call<ChallengeRequest> signIn(@Body SignInRequest request);

	@POST("/api/v2/did/auth")
	Call<AccessCode> auth(@Body ChallengeResponse request);
}

File: src/service/auth/authapi.ts
import {BasePath, BaseService, Body, POST, Response, ResponseTransformer} from 'ts-retrofit';
import {APIResponse} from "../restservice";
import {NotImplementedException} from "../../exceptions";

@BasePath("/api/v2")
export class AuthAPI extends BaseService {
    @POST("/did/signin")
    @ResponseTransformer((data: any, headers?: any) => {
        return APIResponse.handleResponseData(data, (body) => {
            return body['challenge'];
        });
    })
    async signIn(@Body body: object): Promise<Response> {
        throw new NotImplementedException();
    }

    @POST("/did/auth")
    @ResponseTransformer((data: any, headers?: any) => {
        return APIResponse.handleResponseData(data, (body) => {
            return body['token'];
        });
    })
    async auth(@Body body: object): Promise<Response> {
        throw new NotImplementedException();
    }
}
